sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_delete_test_live.cpp

```diff
@@ -353,6 +355,8 @@ TEST_F(KeyVaultKeyClient, RecoverOperationResumeToken)
     auto resumeOperation
         = Azure::Security::KeyVault::Keys::DeleteKeyOperation::CreateFromResumeToken(
             resumeToken, client);
+    // double polling should have no impact on the result
+    resumeOperation.PollUntilDone(m_testPollingIntervalMs);
     resumeOperation.PollUntilDone(m_testPollingIntervalMs);
```

Comment:
This is a pretty good point and test case.

If the caller spawns multiple threads and calls PollUntilDone, the first thread which detects the "completion" should allow all others to continue (rather than failing/crashing).

Is my understanding correct here, @JeffreyRichter?

Reply:
Can you also add a new test case here, to the base `Operation<T>` type, to ensure repeated calls to `PollUntilDone` work correctly:
https://github.com/Azure/azure-sdk-for-cpp/blob/fda88bf7b8389575b9d973821efc604442e10493/sdk/core/azure-core/test/ut/operation_test.cpp#L40-L50

Reply:
you mean something along the lines of the other 100+ LRO tests around ?